frontend/src/lib/v2/DynamicFlow.ts

```diff
@@ -136,23 +340,28 @@ export function getNodeMlmdInfo(
   } else if (NodeTypeNames.SUB_DAG === elem.type) {
     // TODO: Update sub-dag state based on future design.
     const taskName = getTaskKeyFromNodeKey(elem.id);
-    const execution = taskNameToExecution.get(taskName);
-    return { execution };
-  } else {
-    // Edges don't have types yet.
-    // For any element that don't match the above types, copy over directly.
+    const executions = taskNameToExecution
+      .get(taskName)
+      ?.filter(exec => exec.getId() === elem.data?.mlmdId);
+    return executions ? { execution: executions[0] } : {};
   }
   return {};
 }
 
-function getTaskNameToExecution(executions: Execution[]): Map<string, Execution> {
-  const map = new Map<string, Execution>();
+function getTaskNameToExecution(executions: Execution[]): Map<string, Execution[]> {
```

Comment:
(Correct me if I am wrong.) I think the function name `getTaskNameToExecution` is a little confusing especially we are trying do some **set** works. Would it be better if we change the name to `setTaskNameToExecution`? If you think the updated function name is better, can you also change the names for other functions?

Reply:
`setXXX()` is a naming pattern for changing existing object, which means `executions: Execution[]` parameter in this case. However, we are returning `Map<string, Execution[]>` which is the function caller tries to *get*, so I use `getXXX()` pattern here.